### PR TITLE
unordered_dense: add version 3.0.0, improve recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **unordered_dense/***

- add version 3.0.0
- remove wrong `pkg_config_name`
- simplify test_v1_package/CMakeLists.txt

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

File: recipes/unordered_dense/all/conanfile.py [hunk 1]
```diff
@@ -47,9 +42,11 @@ def package_id(self):
...
         if self.info.settings.get_safe("compiler.cppstd"):
             check_min_cppstd(self, self._minimum_cpp_standard)
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version and Version(self.settings.get_safe("compiler.version")) < minimum_version:
-            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+        check_min_vs(self, 192)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.get_safe("compiler.version")) < minimum_version:
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
```
[the remark this answers is not in the document]
Comment (toge) [reply recipes/unordered_dense/all/conanfile.py]: reverted

File: recipes/unordered_dense/all/conanfile.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from conan.tools.build import check_min_cppstd
 from conan.tools.scm import Version
 from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc, check_min_vs
```
Comment (SpaceIm) [inline recipes/unordered_dense/all/conanfile.py:7]:
```suggestion
```

File: recipes/unordered_dense/all/conanfile.py [hunk 3]
```diff
@@ -47,6 +42,7 @@ def package_id(self):
     def validate(self):
         if self.info.settings.get_safe("compiler.cppstd"):
             check_min_cppstd(self, self._minimum_cpp_standard)
+        check_min_vs(self, 192)
```
Comment (SpaceIm) [inline recipes/unordered_dense/all/conanfile.py:45]:
```suggestion
```

File: recipes/unordered_dense/all/conanfile.py [hunk 4]
```diff
@@ -27,17 +28,11 @@ def _minimum_cpp_standard(self):
     @property
     def _compilers_minimum_version(self):
         return {
-            "Visual Studio": "15.7",
-            "msvc": "1914",
             "gcc": "7",
```
Comment (SpaceIm) [inline recipes/unordered_dense/all/conanfile.py:31]:
```suggestion
            "Visual Studio": "15.7",
            "msvc": "191",
            "gcc": "7",
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
this is what I meant

**Comment by toge** [comment]
@SpaceIm 
Thanks a lot!
Now I understand you mean.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SpaceIm — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**